modules/ting/ting.entities.inc

```diff
@@ -272,110 +240,94 @@ class TingEntity extends DingEntity {
    *   List of item creators with inverted fields.
    */
   public function getInvertedCreators() {
-    $this->creator = array();
-    if (!empty($this->reply->record['dc:creator']['oss:sort'])) {
-      $this->creator = $this->reply->record['dc:creator']['oss:sort'];
-    }
-
-    return $this->creator;
+    return empty($this->reply) ? [] : $this->reply->getCreators(TingObjectInterface::NAME_FORMAT_SURNAME_FIRST);
   }
 
-  function getSubjects() {
-    $this->subjects = array();
-    if (!empty($this->reply->record['dc:subject'])) {
-      foreach ($this->reply->record['dc:subject'] as $type => $dc_subject) {
-        if (!in_array($type, array('dkdcplus:genre', 'dkdcplus:DK5', 'dkdcplus:DK5-Text', 'dkdcplus:DBCO', 'dkdcplus:DBCN'))) {
-          $this->subjects = array_merge($this->subjects, $dc_subject);
-        }
-      }
-    }
-    return $this->subjects;
+  /**
+   * Returns list of subjects/keywords for the material.
+   *
+   * @return string[]
+   *   List of subjects, empty if none could be found.
+   */
+  public function getSubjects() {
+    return empty($this->reply) ? [] : $this->reply->getSubjects();
   }
 
-  function getLanguage() {
-    return !empty($this->reply->record['dc:language'][''][0]) ? $this->reply->record['dc:language'][''][0] : FALSE;
+  /**
+   * The language of the material.
```

Comment:
Perhaps it is worth noting here or elsewhere that the language should be human-readable and not a language code.